src/main/java/dev/coms4156/project/individualproject/Interaction.java

```diff
@@ -0,0 +1,246 @@
+          return true;
+        } else {
+          // No interaction found in either order
+          return false;
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      return false;
+    }
+  }
+
+  /**
+   * Get a specific interaction from 2 specified drugs.
+   *
+   * @param drugA  The name of the first drug.
+   * @param drugB  The name of the Second drug.
+   * @return A String indicating the interaction effect between them
+   */
+  public String getInteraction(String drugA, String drugB) {
+    try {
+      // Query where drugA = drugA and drugB = drugB
+      ApiFuture<com.google.cloud.firestore.QuerySnapshot> query = firestore
+          .collection("interactions")
+          .whereEqualTo("drugA", drugA)
+          .whereEqualTo("drugB", drugB)
+          .get();
+
+      com.google.cloud.firestore.QuerySnapshot querySnapshot = query.get();
+
+      if (!querySnapshot.isEmpty()) {
+        // Interaction found
+        com.google.cloud.firestore.DocumentSnapshot document = querySnapshot.getDocuments().get(0);
+        return (String) document.get("interactionEffect");
+      } else {
+        // Try the reverse (drugA = drugB and drugB = drugA)
+        ApiFuture<com.google.cloud.firestore.QuerySnapshot> reverseQuery = firestore
+            .collection("interactions")
+            .whereEqualTo("drugA", drugB)
+            .whereEqualTo("drugB", drugA)
+            .get();
+
+        com.google.cloud.firestore.QuerySnapshot reverseQuerySnapshot = reverseQuery.get();
+
+        if (!reverseQuerySnapshot.isEmpty()) {
+          // Interaction found with reverse order
+          com.google.cloud.firestore.DocumentSnapshot reverseDocument = reverseQuerySnapshot
+              .getDocuments().get(0);
+          return (String) reverseDocument.get("interactionEffect");
+        } else {
+          // No interaction found in either order
+          return "No known interaction between " + drugA + " and " + drugB;
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      return "Error retrieving drug interaction: " + e.getMessage();
+    }
+  }
+
+  /**
+   * Get a specific interaction from a list of specified drugs.
+   *
+   * @param drugs  The List of drugs to check for any pairwise interaction.
+   * @return A List of String indicating all pairwise interaction effect
+   */
+  public List<String> getInteraction(List<String> drugs) {
+    List<String> interactions = new ArrayList<>();
+
+    try {
+      for (int i = 0; i < drugs.size(); i++) {
+        for (int j = i + 1; j < drugs.size(); j++) {
+          String drugA = drugs.get(i);
+          String drugB = drugs.get(j);
+
+          String interaction = getInteraction(drugA, drugB);
+          interactions.add(interaction);
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      interactions.add("Error retrieving drug interactions: " + e.getMessage());
+    }
+
```

Comment:
Currently, the `getInteraction(List<String> drugs)` method in the `Interaction` service returns a list of strings, which doesn't provide enough structure for our API response (check the spec doc). For example, it might return something like this:

```java
[
    "Increased risk of bleeding.",
    "No known interaction between Warfarin and Ibuprofen",
    "No known interaction between Warfarin and Sudafed",
    "No known interaction between Warfarin and Benadryl",
    "Increased risk of gastrointestinal bleeding.",
    "No known interaction between Aspirin and Sudafed",
    "No known interaction between Aspirin and Benadryl",
    "No known interaction between Ibuprofen and Sudafed",
    "No known interaction between Ibuprofen and Benadryl",
    "No known interaction between Sudafed and Benadryl"
]
```

I think we should modify this to return more detailed information. For example you could change the `Interaction` service to return a list of structured objects in a small class or somethign else idk